docs/modules/ROOT/pages/explanations/decisions/forgejo.adoc

```diff
@@ -0,0 +1,402 @@
+|See <<Kubernetes Resource Configuration, below>>
+
+|===
+
+=== Plan Upgrade / Downgrade
+
+Plans can be changed anytime, as long as the database stays the same.
+Changing between plans with different databases is currently not supported.
+It will be offered once the process of changing databases is automated.
+
+== Forgejo Configuration
+
+In this section the Forgejo configuration decisions are documented.
+
+For Forgejo by VSHN the user can choose various options themselves, but we set reasonable defaults, as documented in this section.
+In Codey, the choices are given by the <<Codey Plans, plan>>.
+
+=== Database
+
+There are several options for the database, most prominently PostgreSQL and SQLite.
+
+Default:: https://products.vshn.ch/appcat/postgresql.html[PostgreSQL by VSHN^]
+Offered Choices:: PostgreSQL, SQLite
+
+Migration between different databases can be done with the https://forgejo.org/docs/latest/admin/command-line/#dump[`forgejo dump`] command.
+
+Ref: https://forgejo.org/docs/latest/admin/recommendations/#databasedb_type[Recommended Settings and Tips: Database^]
+
+=== Storage
+
+We use local storage for everything, therefore a PVC is mounted to `/data`.
+
+While we could also use object storage for most of the data, it complicates a lot.
+For example backup would become an additional burden, as well as storage space monitoring also won't be that easy anymore.
+And it's not possible to store Git repositories on object storage, this way we'd have multiple storage locations.
+
+Ref: https://forgejo.org/docs/latest/admin/storage/[Admin documentation: Storage^]
+
+=== Cache
+
+There are several options for caching, most prominently Redis and TwoQueue.
+
+Default:: https://products.vshn.ch/appcat/redis.html[Redis by VSHN^]
+Offered Choices:: Redis, `twoqueue`
+
+`twoqueue` uses the recommended configuration for the `HOST` parameter:
+
+----
+{"size":100, "recent_ratio":0.25, "ghost_ratio":0.5} 
+----
+
+Refs:
+
+* https://forgejo.org/docs/latest/admin/config-cheat-sheet/#cache-cache[Configuration Cheat Sheet: Cache^]
+* https://forgejo.org/docs/latest/admin/recommendations/#cacheadapter[Recommended Settings and Tips: Cache^]
+
+=== Queue
+
+There are several options for queuing, most prominently Redis and https://github.com/google/leveldb[LevelDB^].
+
+Default:: https://products.vshn.ch/appcat/redis.html[Redis by VSHN^]
+Offered Choices:: Redis, `level`
+
+Ref: https://forgejo.org/docs/latest/admin/config-cheat-sheet/#queue-queue-and-queue[Configuration Cheat Sheet: Queue^]
+
+== Backup and Restore
+
+Two mechanisms are used for Backup:
+
+File-based data::
+K8up regularly stores the content of `/data` in the backup destination.
+Restoring happens with K8up mechanisms to get the files back into the `/data` directory.
+
+Database::
+We'll use https://forgejo.org/docs/latest/admin/command-line/#dump[`forgejo dump`] as a https://docs.k8up.io/k8up/2.11/how-tos/application-aware-backups.html[backup command^] for K8up.
```

Comment:
`forgejo dump` doesn't just backup the database, it makes a dump of everything. This will also ensure that the state between db and repositories is consistent.

So I wouldn't differentiate between file-based and database backup.

The dump command even supports dumping to stdout, which is very neat. However, the dump command will still need some additional temp space during the dumping (I guess these temp files will be the same size as the whole instance).

Reply:
Good input, I amended the whole backup and restore section accordingly.